### PR TITLE
Add optExtra methods to PwsResult and UrlDevice

### PR DESCRIPTION
**Pull request by cco3**

### CODE REVIEW

File: java/libs/src/main/java/org/physical_web/collection/UrlDevice.java [hunk 1]
```diff
@@ -159,49 +159,153 @@ public String getUrl() {
...
     return mExtraData.optJSONArray(key);
   }
 
   /**
    * Get extra JSONObject value.
-   * @return The stored value or null if it doesn't exist in specified form.
+   * @param key The key of the stored value.
+   * @return The stored value.
    */
   public JSONObject getExtraJSONObject(String key) {
```
Comment (nondebug) [inline java/libs/src/main/java/org/physical_web/collection/UrlDevice.java]: addExtra(String, Object) can add "Any object of type JSONObject, JSONArray, String, Boolean, Integer, Long, Double, NULL, or null. May not be NaNs or infinities" but currently we have no way to retrieve a string value.

File: java/libs/src/main/java/org/physical_web/collection/PwsResult.java [hunk 2]
```diff
@@ -302,49 +302,153 @@ public String getGroupId() {
 
   /**
    * Get extra boolean value.
+   * @param key The key of the stored value.
    * @return The stored value or null if it doesn't exist in specified form.
```
Comment (nondebug) [inline java/libs/src/main/java/org/physical_web/collection/PwsResult.java]: The getExtra.* methods should say we'll throw JSONException if the key isn't present

**Comment by cco3** [comment]
Updated

**Comment by nondebug** [comment]
lgtm

**Pull request merged**